### ISSUE TITLE
Improve code style

### ISSUE THREAD

**Issue by wlandau-lilly** [wlandau-lilly/drake#40]
Should use `formatR::tidy_dir()` and `lintr::lint_package()` to conform to the [style guide](http://adv-r.had.co.nz/Style.html).

**Comment by wlandau-lilly** [comment]
`tidy_source()` and `tidy_dir()` remove many complaints from `lint_package()`, but not all of them, and I think the output ends up less readable in some ways. If I were to fix the style, I would need to manually go through each source file, example, and vignette regardless of how I preprocess with `formatR`. I will do this if someone expresses a strong enough preference, but otherwise, I do not think it is worth the time.

**Comment by wlandau-lilly** [comment]
I think it's worth keeping this one on the table. It would be enough to follow [Hadley's style guide](http://adv-r.had.co.nz/Style.html).

**Comment by wlandau-lilly** [comment]
I would add this issue to the next CRAN milestone, but I am not sure I will work on it myself until #21 is fixed.

**Comment by wlandau-lilly** [comment]
Also, `dataframes_graph()` is getting pretty big and ugly. But again, I probably will not work on it until #21 is fixed.

**Comment by AlexAxthelm** [comment]
I can start on this, if you want. Spacing/formatting is pretty easy for me, and my editor runs `lintr` automatically, so this would actually help me spot errors in code, rather than lint warnings. 😄 

I know you have some changes waiting to merge (re: #21 / # 22 ), but I can start with any files that you haven't changed recently, and then once `master` is up to date, run through the rest.

**Comment by wlandau-lilly** [comment]
@AlexAxthelm that would be an amazing help! Here are the files safe to modify in the `R` folder:

- clean.R
- console.R
- current.R
- debug.R
- dependencies.R
- deprecate.R
- envir.R
- find.R
- generate.R
- graph.R
- lapply.R
- Makefile.R
- make.R
- mclapply.R
- outdated.R
- plan.R
- progress.R
- read.R
- test.R
- timestamp.R
- zzz.R

and in `tests/testthat`:

- test-command-changes.R
- test-envir.R
- test-examples.R
- test-full-build.R
- test-generate.R
- test-import-file.R
- test-import-object.R
- test-intermediate-file.R
- test-Makefile.R
- test-namespaced.R
- test-other-features.R

And `inst/examples/basic/basic.R` is also fair game if you work in a mention of `build_times()`.

**Comment by AlexAxthelm** [comment]
No problem. I’ll get started on these soon.

**Comment by wlandau-lilly** [comment]
Great! This one seems more straight-forward than your other pull requests, so I will probably accept it first. Feel free to add yourself as a contributor in the DESCRIPTION.

By the way, my own changes waiting to merge were all super easy, and I think I spelled out all the details in the issue tracker. If all/most of the fixes happen to come in via pull request, I will accept them and overwrite my company's internal remote. It takes my company several weeks to let me release any changes myself. Currently working to change that.

**Comment by wlandau-lilly** [comment]
In addition, the only colleague authorized to approve my code is on vacation this week.

**Comment by wlandau-lilly** [comment]
Would it be too much to eventually [integrate Travis and lintr](https://github.com/jimhester/lintr/blob/master/README.md#travis-ci)? It would be handy, but it is a high bar for most contributors.

**Comment by AlexAxthelm** [comment]
I'll make this the last change, after the other PRs come through. I'll set it for [non-failing lints](https://github.com/jimhester/lintr/blob/master/README.md#non-failing-lints)

**Comment by wlandau-lilly** [comment]
Thanks!

**Comment by wlandau-lilly** [comment]
Edit: to fix #65, I modified
- `make.R`
- `test-import-file.R`
- `test-other-features.R`
These may have merge conflicts.

**Comment by wlandau-lilly** [comment]
Also removing `debug.R` and `test.R` from the safe list because I fixed #58 on my end. Thanks to the code from #59, there was not actually much to do. Anything incoming that uses `testrun(config)` in tests should still work as expected.

**Comment by AlexAxthelm** [comment]
Awesome. I'll probably submit fixes over a few PRs, to keep them more atomic.

**Comment by wlandau-lilly** [comment]
Sounds great! 

By the way, I think we should permit camel case in `lint()` since visNetwork clings to it (understandably so, coming from JavaScript).

**Comment by AlexAxthelm** [comment]
If that's what you want. My inclination would be to `# nolint` those lines though, since it isn't actually called all that much.

**Comment by wlandau-lilly** [comment]
Right, I forgot about that. I will go ahead and `# nolint` the 13 lines of `plotgraph.R` that have camel case, and we can leave the default linter alone. Both lintr and withr are new to me, and I don't have the right habits yet.

**Comment by wlandau-lilly** [comment]
Current summary of lints in my local development copy using Jim Hester's `summary.lints()` from jimhester/lintr#260.

```r
                                 filenames style warning error
1           tests/testthat/test-generate.R    86       0     0
2                             R/generate.R    74       0     0
3              inst/examples/basic/basic.R    52       0     0
4                             R/Makefile.R    35       0     0
5                                 R/read.R    34       0     0
6           tests/testthat/test-Makefile.R    34       0     0
7              tests/testthat/test-envir.R    32       0     0
8                             R/progress.R    31       0     0
9                                R/graph.R    29       0     0
10                                R/plan.R    27       0     0
11                                R/make.R    21       0     0
12                            R/outdated.R    20       0     0
13     tests/testthat/test-import-object.R    18       0     0
14        tests/testthat/test-namespaced.R    17       0     0
15                           R/timestamp.R    16       0     0
16        tests/testthat/test-full-build.R    16       0     0
17   tests/testthat/test-command-changes.R    13       0     0
18                                R/find.R     7       0     0
19 tests/testthat/test-intermediate-file.R     7       0     0
```

**Comment by AlexAxthelm** [comment]
That's not bad at all. I'll get to it as soon as I finish a side feature that I'm working on. Any of those that might give me merge conflicts?

**Comment by wlandau-lilly** [comment]
Unfortunately, yes. In fact, I just realized all the testing files will have merge conflicts because I replaced `test_that()` with `test_with_dir()`. Here are the files that are safe to change.

- `R/generate.R`
- `R/read.R`
- `R/progress.R`
- `R/graph.R`
- `R/outdated.R`
- `R/plan.R`

 I will take care of the rest myself.

Unfortunately, it looks like I will have to resolve a merge conflict in `R/parallel.R` before I push the next release anyway. I hope that is the only one.

**Comment by AlexAxthelm** [comment]
Okay. If you want to leave the `test` lints until after 4.1.0, I can take care of them.

**Comment by wlandau-lilly** [comment]
Thank you, that would be great.

**Comment by wlandau-lilly** [comment]
For completeness, here are the files in the queue for linting after 4.1.0:

```r
                                filenames style warning error
              inst/examples/basic/basic.R    52       0     0
                                 R/find.R     7       0     0
                             R/Makefile.R    35       0     0
                                 R/make.R    21       0     0
                            R/timestamp.R    16       0     0
    tests/testthat/test-command-changes.R    13       0     0
	      tests/testthat/test-envir.R    32       0     0
         tests/testthat/test-full-build.R    16       0     0
           tests/testthat/test-generate.R    86       0     0
      tests/testthat/test-import-object.R    18       0     0
  tests/testthat/test-intermediate-file.R     7       0     0
           tests/testthat/test-Makefile.R    34       0     0
         tests/testthat/test-namespaced.R    17       0     0
```

**Comment by wlandau-lilly** [comment]
Fixed in a767a35d369b2708c20a8447b2a56baf4c636ae1, along with most of the code coverage. @AlexAxthelm if you still want to clean up the code some more to make it conform to Vim standards, I will accept those PRs. But I think the style is now good enough to close this issue.
